components/authentication-framework/org.wso2.carbon.identity.application.authentication.endpoint/src/main/webapp/login.jsp

```diff
@@ -141,8 +142,13 @@
         </div>
         <% } %>
 
-        <form action="../commonauth" method="post" id="loginForm" class="form-horizontal">
-                    <%
+        <% String webContextRoot = ServerConfiguration.getInstance().getFirstProperty(Constants.WEB_CONTEXT_ROOT);
+            if (StringUtils.isNotBlank(webContextRoot)) { %>
+        <form action="../<%=webContextRoot%>/commonauth" method="post" id="loginForm" class="form-horizontal">
+                    <% }else{%>
+            <form action="../commonauth" method="post" id="loginForm" class="form-horizontal">
+                        <%}%>
+                        <%
```

Comment:
No need to do this change. If user adds WebContextRoot he also has to change this JSP manually. We can't use ServerConfiguration inside this webapp because we assume this webapp will be taken outside of IS server and used. Therefore we can't have any carbon dependencies here.